packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/sign_in_state_machine.dart

```diff
@@ -630,6 +651,56 @@ final class SignInStateMachine
     }
   }
 
+  /// Handles the MFA setup challenge.
+  @protected
+  Future<RespondToAuthChallengeRequest?> handleMfaSetup({
+    SignInEvent? event,
+    required bool hasUserResponse,
+  }) async {
+    final allowedMfaTypes = _allowedMfaTypes;
+    // Exclude MfaType.sms from consideration
+    final mfaTypesForSetup = allowedMfaTypes?.difference({MfaType.sms});
+    if (mfaTypesForSetup == null || mfaTypesForSetup.isEmpty) {
+      throw const InvalidUserPoolConfigurationException(
+        'No eligible MFA types are allowed for setup.',
+        recoverySuggestion: 'Check your user pool MFA configuration.',
+      );
+    }
+
+    if (mfaTypesForSetup.length == 1) {
+      final mfaType = mfaTypesForSetup.first;
+      if (mfaType == MfaType.totp) {
+        _enableMfaType = MfaType.totp;
+        _totpSetupResult ??= await associateSoftwareToken();
+        if (hasUserResponse) {
+          return createMfaSetupRequest(event as SignInRespondToChallenge);
+        } else {
+          // Need to prompt user for the TOTP code
+          return null;
+        }
+      } else if (mfaType == MfaType.email) {
+        _enableMfaType = MfaType.email;
+        if (hasUserResponse) {
+          return createEmailMfaSetupRequest(event as SignInRespondToChallenge);
+        } else {
+          // Need to prompt user for the email verification code
+          return null;
+        }
+      } else {
+        throw InvalidUserPoolConfigurationException(
+          'Unsupported MFA type: ${mfaType.name}',
+          recoverySuggestion: 'Check your user pool MFA configuration.',
+        );
+      }
+    } else if (hasUserResponse) {
+      // Handle user's selection
+      return createMfaSetupRequest(event as SignInRespondToChallenge);
```

Comment:
instead of nested if else statement move this to the top like:
```
if (mfaTypesForSetup.length >1) {
  return createMfaSetupRequest(event as SignInRespondToChallenge);
}
if (mfaTypesForSetup.length == 1) {
...
}
```
also please clean up the nested if else statements  and use if or switch as the if statement as the conditions are mutally exclusive